=========
InfoScan
=========

A software to :red:`I`\ dentify :red:`N`\ ovel transcripts :red:`F`\ r\ :red:`O`\ m :red:`S`\ :red:`C`\ RNA-Seq data and sp:red:`A`\ tial tra:red:`N`\ scriptome data.
=======================================================================================================================================================================================

.. image:: images/main.png

      In this study, a multi-platform visualization software **InfoScan** was developed, which can accurately identify existing and new transcripts in single-cell RNA-Seq data, and evaluate their coding ability by machine learning method. In addition, InfoScan can also integrate the information of the spatial transcriptome to elucidate the spatiotemporal specificity of these transcripts. InfoScan is dedicated to mining the cell-specific profile of these novel transcripts, and performs grouping comparisons and cell type inference based on mRNA and long non-coding RNA, respectively. In addition, we can also predict the function of novel transcripts in cells based on co-expression analysis.

The flow chart below depicts the different tool modules that are currently available.

.. image:: images/start_workflow.jpg

Contents:
---------
.. toctree::
   :maxdepth: 1

   content/installation
   content/InfoUpload
   content/InfoAssembly
   content/NovelScan
   content/CellInfo
   content/FuncScan
   content/SpatialScan
   content/about


.. tip:: For support or questions please post to `Biostars <http://biostars.org>`__. For bug reports and feature requests please open an issue on `github <https://github.com/worsteggs/InfoScan_readthedocs/issues>`__.



.. .. image:: images/logo_mpi-ie.jpg

.. This tool suite is developed by the `Bioinformatics Facility <http://www1.ie-freiburg.mpg.de/bioinformaticsfac>`_ at the `Max Planck Institute for Immunobiology and Epigenetics, Freiburg <http://www1.ie-freiburg.mpg.de/>`_.
