<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>bamCoverage &mdash; InfoScan 0.1 documentation</title>
      <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/css/custom-style.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
        <script src="../../_static/jquery.js"></script>
        <script src="../../_static/underscore.js"></script>
        <script src="../../_static/doctools.js"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../../index.html" class="icon icon-home"> InfoScan
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../InfoUpload.html">InfoUpload</a></li>
<li class="toctree-l1"><a class="reference internal" href="../NovelScan.html">NovelScan</a></li>
<li class="toctree-l1"><a class="reference internal" href="../CellInfo.html">CellInfo</a></li>
<li class="toctree-l1"><a class="reference internal" href="../about.html">About</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">InfoScan</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>bamCoverage</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/content/tools/bamCoverage.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="bamcoverage">
<h1>bamCoverage<a class="headerlink" href="#bamcoverage" title="Permalink to this headline"></a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#usage-hints" id="id1">Usage hints</a></p></li>
<li><p><a class="reference internal" href="#usage-example-for-chip-seq" id="id2">Usage example for ChIP-seq</a></p></li>
<li><p><a class="reference internal" href="#usage-examples-for-rna-seq" id="id3">Usage examples for RNA-seq</a></p>
<ul>
<li><p><a class="reference internal" href="#regular-bigwig-track" id="id4">Regular bigWig track</a></p></li>
<li><p><a class="reference internal" href="#separate-tracks-for-each-strand" id="id5">Separate tracks for each strand</a></p>
<ul>
<li><p><a class="reference internal" href="#versions-before-2-2" id="id6">Versions before 2.2</a></p></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<img alt="images/norm_IGVsnapshot_indFiles.png" src="images/norm_IGVsnapshot_indFiles.png" />
<p>If you are not familiar with BAM, bedGraph and bigWig formats, you can read up on that in our <a class="reference internal" href="../help_glossary.html"><span class="doc">Glossary of NGS terms</span></a></p>
<section id="usage-hints">
<h2><a class="toc-backref" href="#id1">Usage hints</a><a class="headerlink" href="#usage-hints" title="Permalink to this headline"></a></h2>
<ul class="simple">
<li><p>A smaller bin size value will result in a higher resolution of the coverage track but also in a larger file size.</p></li>
<li><p>The 1x normalization (RPGC) requires the input of a value for the <strong>effective genome size</strong>, which is the mappable part of the reference genome. Of course, this value is species-specific. The command line help of this tool offers suggestions for a number of model species.</p></li>
<li><p>It might be useful for some studies to exclude certain chromosomes in order to avoid biases, e.g. chromosome X, as male mice contain a pair of each autosome, but usually only a single X chromosome.</p></li>
<li><p>By default, the read length is <strong>NOT</strong> extended! This is the preferred setting for <strong>spliced-read</strong> data like RNA-seq, where one usually wants to rely on the detected read locations only. A read extension would neglect potential splice sites in the unmapped part of the fragment.
Other data, e.g. Chip-seq, where fragments are known to map contiguously, should be processed with read extension (<code class="docutils literal notranslate"><span class="pre">--extendReads</span> <span class="pre">[INTEGER]</span></code>).</p></li>
<li><p>For paired-end data, the fragment length is generally defined by the two read mates. The user provided fragment length is only used as a fallback for singletons or mate reads that map too far apart (with a distance greater than four times the fragment length or are located on different chromosomes).</p></li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If you already normalized for GC bias using <code class="docutils literal notranslate"><span class="pre">correctGCbias</span></code>, you should absolutely <strong>NOT</strong> set the parameter <code class="docutils literal notranslate"><span class="pre">--ignoreDuplicates</span></code>!</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Like BAM files, bigWig files are compressed, binary files. If you would like to see the coverage values, choose the bedGraph output via <code class="docutils literal notranslate"><span class="pre">--outFileFormat</span></code>.</p>
</div>
</section>
<section id="usage-example-for-chip-seq">
<h2><a class="toc-backref" href="#id2">Usage example for ChIP-seq</a><a class="headerlink" href="#usage-example-for-chip-seq" title="Permalink to this headline"></a></h2>
<p>This is an example for ChIP-seq data using additional options (smaller bin size for higher resolution, normalizing coverage to 1x mouse genome size, excluding chromosome X during the normalization step, and extending reads):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>bamCoverage --bam a.bam -o a.SeqDepthNorm.bw <span class="se">\</span>
    --binSize <span class="m">10</span>
    --normalizeUsing RPGC
    --effectiveGenomeSize <span class="m">2150570000</span>
    --ignoreForNormalization chrX
    --extendReads
</pre></div>
</div>
<p>If you had run the command with <code class="docutils literal notranslate"><span class="pre">--outFileFormat</span> <span class="pre">bedgraph</span></code>, you could easily peak into the resulting file.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ head SeqDepthNorm_chr19.bedgraph
<span class="m">19</span>  <span class="m">60150</span>   <span class="m">60250</span>   <span class="m">9</span>.32
<span class="m">19</span>  <span class="m">60250</span>   <span class="m">60450</span>   <span class="m">18</span>.65
<span class="m">19</span>  <span class="m">60450</span>   <span class="m">60650</span>   <span class="m">27</span>.97
<span class="m">19</span>  <span class="m">60650</span>   <span class="m">60950</span>   <span class="m">37</span>.29
<span class="m">19</span>  <span class="m">60950</span>   <span class="m">61000</span>   <span class="m">27</span>.97
<span class="m">19</span>  <span class="m">61000</span>   <span class="m">61050</span>   <span class="m">18</span>.65
<span class="m">19</span>  <span class="m">61050</span>   <span class="m">61150</span>   <span class="m">27</span>.97
<span class="m">19</span>  <span class="m">61150</span>   <span class="m">61200</span>   <span class="m">18</span>.65
<span class="m">19</span>  <span class="m">61200</span>   <span class="m">61300</span>   <span class="m">9</span>.32
<span class="m">19</span>  <span class="m">61300</span>   <span class="m">61350</span>   <span class="m">18</span>.65
</pre></div>
</div>
<p>As you can see, each row corresponds to one region. If consecutive bins have the same number of reads overlapping, they will be merged.</p>
</section>
<section id="usage-examples-for-rna-seq">
<h2><a class="toc-backref" href="#id3">Usage examples for RNA-seq</a><a class="headerlink" href="#usage-examples-for-rna-seq" title="Permalink to this headline"></a></h2>
<p>Note that some BAM files are filtered based on SAM flags (<a class="reference external" href="https://broadinstitute.github.io/picard/explain-flags.html">Explain SAM flags</a>).</p>
<section id="regular-bigwig-track">
<h3><a class="toc-backref" href="#id4">Regular bigWig track</a><a class="headerlink" href="#regular-bigwig-track" title="Permalink to this headline"></a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>bamCoverage -b a.bam -o a.bw
</pre></div>
</div>
</section>
<section id="separate-tracks-for-each-strand">
<h3><a class="toc-backref" href="#id5">Separate tracks for each strand</a><a class="headerlink" href="#separate-tracks-for-each-strand" title="Permalink to this headline"></a></h3>
<p>Sometimes it makes sense to generate two independent <a class="reference internal" href="../help_glossary.html#bigwig"><span class="std std-ref">bigWig</span></a> files for all reads on the forward and reverse strand, respectively.
As of deepTools version 2.2, one can simply use the <code class="docutils literal notranslate"><span class="pre">--filterRNAstrand</span></code> option, such as <code class="docutils literal notranslate"><span class="pre">--filterRNAstrand</span> <span class="pre">forward</span></code> or <code class="docutils literal notranslate"><span class="pre">--filterRNAstrand</span> <span class="pre">reverse</span></code>.
This handles paired-end and single-end datasets. For older versions of deepTools, please see the instructions below.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The <code class="docutils literal notranslate"><span class="pre">--filterRNAstrand</span></code> option assumes the sequencing library generated from ILLUMINA dUTP/NSR/NNSR methods, which are the most commonly used method for
library preparation, where Read 2 (R2) is in the direction of RNA strand (<strong>reverse-stranded</strong> library). However other methods exist, which generate read
R1 in the direction of RNA strand (<a class="reference external" href="http://www.nature.com/nmeth/journal/v7/n9/full/nmeth.1491.html">see this review</a>). For these libraries,
<code class="docutils literal notranslate"><span class="pre">--filterRNAstrand</span></code> will have an opposite behavior, i.e. <code class="docutils literal notranslate"><span class="pre">--filterRNAstrand</span> <span class="pre">forward</span></code> will give you reverse strand signal and vice-versa.</p>
</div>
<section id="versions-before-2-2">
<h4><a class="toc-backref" href="#id6">Versions before 2.2</a><a class="headerlink" href="#versions-before-2-2" title="Permalink to this headline"></a></h4>
<p>To follow the examples, you need to know that <code class="docutils literal notranslate"><span class="pre">-f</span></code> will tell <code class="docutils literal notranslate"><span class="pre">samtools</span> <span class="pre">view</span></code> to <strong>include</strong> reads with the indicated flag, while <code class="docutils literal notranslate"><span class="pre">-F</span></code> will lead to the <strong>exclusion</strong> of reads with the respective flag.</p>
<p><strong>For a stranded `single-end` library</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Forward strand</span>
bamCoverage -b a.bam -o a.fwd.bw --samFlagExclude <span class="m">16</span>

<span class="c1"># Reverse strand</span>
bamCoverage -b a.bam -o a.rev.bw --samFlagInclude <span class="m">16</span>
</pre></div>
</div>
<p><strong>For a stranded `paired-end` library</strong></p>
<p>Now, this gets a bit cumbersome, but future releases of deepTools will make this more straight-forward.
For now, bear with us and perhaps read up on SAM flags, e.g. <a class="reference external" href="http://ppotato.wordpress.com/2010/08/25/samtool-bitwise-flag-paired-reads/">here</a>.</p>
<p>For paired-end samples, we assume that a proper pair should have the mates on opposing strands where the Illumina strand-specific protocol produces reads in a <code class="docutils literal notranslate"><span class="pre">R2-R1</span></code> orientation. We basically follow the recipe given <a class="reference external" href="https://www.biostars.org/p/92935/">in this biostars tutorial</a>.</p>
<p>To get the file for transcripts that originated from the <strong>forward strand</strong>:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># include reads that are 2nd in a pair (128);</span>
<span class="c1"># exclude reads that are mapped to the reverse strand (16)</span>
$ samtools view -b -f <span class="m">128</span> -F <span class="m">16</span> a.bam &gt; a.fwd1.bam

<span class="c1"># exclude reads that are mapped to the reverse strand (16) and</span>
<span class="c1"># first in a pair (64): 64 + 16 = 80</span>
$ samtools view -b -f <span class="m">80</span> a.bam &gt; a.fwd2.bam

<span class="c1"># combine the temporary files</span>
$ samtools merge -f fwd.bam a.fwd1.bam a.fwd2.bam

<span class="c1"># index the filtered BAM file</span>
$ samtools index fwd.bam

<span class="c1"># run bamCoverage</span>
$ bamCoverage -b fwd.bam -o a.fwd.bigWig

<span class="c1"># remove the temporary files</span>
$ rm a.fwd*.bam
</pre></div>
</div>
<p>To get the file for transcripts that originated from the <strong>reverse strand</strong>:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># include reads that map to the reverse strand (128)</span>
<span class="c1"># and are second in a pair (16): 128 + 16 = 144</span>
$ samtools view -b -f <span class="m">144</span> a.bam &gt; a.rev1.bam

<span class="c1"># include reads that are first in a pair (64), but</span>
<span class="c1"># exclude those ones that map to the reverse strand (16)</span>
$ samtools view -b -f <span class="m">64</span> -F <span class="m">16</span> a.bam &gt; a.rev2.bam

<span class="c1"># merge the temporary files</span>
$ samtools merge -f rev.bam a.rev1.bam a.rev2.bam

<span class="c1"># index the merged, filtered BAM file</span>
$ samtools index rev.bam

<span class="c1"># run bamCoverage</span>
$ bamCoverage -b rev.bam -o a.rev.bw

<span class="c1"># remove temporary files</span>
$ rm a.rev*.bam
</pre></div>
</div>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Zilch.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>