<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>multiBigwigSummary &mdash; InfoScan 0.1 documentation</title>
      <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/css/custom-style.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
        <script src="../../_static/jquery.js"></script>
        <script src="../../_static/underscore.js"></script>
        <script src="../../_static/doctools.js"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../../index.html" class="icon icon-home"> InfoScan
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../InfoUpload.html">InfoUpload</a></li>
<li class="toctree-l1"><a class="reference internal" href="../NovelScan.html">NovelScan</a></li>
<li class="toctree-l1"><a class="reference internal" href="../CellInfo.html">CellInfo</a></li>
<li class="toctree-l1"><a class="reference internal" href="../about.html">About</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">InfoScan</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>multiBigwigSummary</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/content/tools/multiBigwigSummary.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="multibigwigsummary">
<h1>multiBigwigSummary<a class="headerlink" href="#multibigwigsummary" title="Permalink to this headline"></a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#example" id="id1">Example</a></p></li>
<li><p><a class="reference internal" href="#multibigwigsummary-in-galaxy" id="id2">multiBigwigSummary in Galaxy</a></p></li>
</ul>
</div>
<section id="example">
<h2><a class="toc-backref" href="#id1">Example</a><a class="headerlink" href="#example" title="Permalink to this headline"></a></h2>
<p>In the following example, the average values for our test ENCODE
ChIP-Seq datasets are computed for consecutive genome bins (default size: 10kb) by using the <cite>bins</cite> mode.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ deepTools2.0/bin/multiBigwigSummary bins <span class="se">\</span>
 -b testFiles/H3K4Me1.bigWig testFiles/H3K4Me3.bigWig testFiles/H3K27Me3.bigWig testFiles/Input.bigWig <span class="se">\</span>
 --labels H3K4me1 H3K4me3 H3K27me3 input <span class="se">\</span>
 -out scores_per_bin.npz --outRawCounts scores_per_bin.tab

$ head scores_per_bin.tab
    <span class="c1">#&#39;chr&#39;  &#39;start&#39; &#39;end&#39;   &#39;H3K4me1&#39;       &#39;H3K4me3&#39;       &#39;H3K27me3&#39;      &#39;input&#39;</span>
    <span class="m">19</span>      <span class="m">0</span>       <span class="m">10000</span>   <span class="m">0</span>.0     <span class="m">0</span>.0     <span class="m">0</span>.0     <span class="m">0</span>.0
    <span class="m">19</span>      <span class="m">10000</span>   <span class="m">20000</span>   <span class="m">0</span>.0     <span class="m">0</span>.0     <span class="m">0</span>.0     <span class="m">0</span>.0
    <span class="m">19</span>      <span class="m">20000</span>   <span class="m">30000</span>   <span class="m">0</span>.0     <span class="m">0</span>.0     <span class="m">0</span>.0     <span class="m">0</span>.0
    <span class="m">19</span>      <span class="m">30000</span>   <span class="m">40000</span>   <span class="m">0</span>.0     <span class="m">0</span>.0     <span class="m">0</span>.0     <span class="m">0</span>.0
    <span class="m">19</span>      <span class="m">40000</span>   <span class="m">50000</span>   <span class="m">0</span>.0     <span class="m">0</span>.0     <span class="m">0</span>.0     <span class="m">0</span>.0
    <span class="m">19</span>      <span class="m">50000</span>   <span class="m">60000</span>   <span class="m">0</span>.0221538461538 <span class="m">0</span>.0     <span class="m">0</span>.00482142857143        <span class="m">0</span>.0522717391304
    <span class="m">19</span>      <span class="m">60000</span>   <span class="m">70000</span>   <span class="m">4</span>.27391282051   <span class="m">1</span>.625   <span class="m">0</span>.634116071429  <span class="m">1</span>.29124347826
    <span class="m">19</span>      <span class="m">70000</span>   <span class="m">80000</span>   <span class="m">13</span>.0891675214   <span class="m">24</span>.65   <span class="m">1</span>.8180625       <span class="m">2</span>.80073695652
    <span class="m">19</span>      <span class="m">80000</span>   <span class="m">90000</span>   <span class="m">1</span>.74591965812   <span class="m">0</span>.29    <span class="m">4</span>.35576785714   <span class="m">0</span>.92987826087
</pre></div>
</div>
<p>To compute the average values for a set of genes, use the <cite>BED-file</cite> mode.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ deepTools2.0/bin/multiBigwigSummary BED-file <span class="se">\</span>
 --bwfiles testFiles/*bigWig <span class="se">\</span>
 --BED testFiles/genes.bed <span class="se">\</span>
 --labels H3K27me3 H3K4me1 H3K4me3 HeK9me3 input <span class="se">\</span>
 -out scores_per_transcript.npz --outRawCounts scores_per_transcript.tab

 $ head scores_per_transcript.tab
 <span class="c1">#&#39;chr&#39;     &#39;start&#39; &#39;end&#39;   &#39;H3K27me3&#39;      &#39;H3K4me1&#39;       &#39;H3K4me3&#39;       &#39;HeK9me3&#39;       &#39;input&#39;</span>
<span class="m">19</span>  <span class="m">60104</span>   <span class="m">70951</span>   <span class="m">0</span>.663422099099  <span class="m">4</span>.37103606574   <span class="m">14</span>.9609108509   <span class="m">0</span>.596631607217  <span class="m">1</span>.34274297191
<span class="m">19</span>  <span class="m">60950</span>   <span class="m">70966</span>   <span class="m">0</span>.714223982699  <span class="m">4</span>.54650763906   <span class="m">16</span>.2336261981   <span class="m">0</span>.62173674295   <span class="m">1</span>.41719308888
<span class="m">19</span>  <span class="m">62114</span>   <span class="m">70944</span>   <span class="m">0</span>.747578769617  <span class="m">4</span>.84009060023   <span class="m">18</span>.2951302378   <span class="m">0</span>.648723472352  <span class="m">1</span>.51324474371
<span class="m">19</span>  <span class="m">63820</span>   <span class="m">70951</span>   <span class="m">0</span>.781816722009  <span class="m">5</span>.30500631048   <span class="m">22</span>.5579862572   <span class="m">0</span>.682862029229  <span class="m">1</span>.55490104062
<span class="m">19</span>  <span class="m">65057</span>   <span class="m">66382</span>   <span class="m">0</span>.528301886792  <span class="m">5</span>.45886792453   <span class="m">0</span>.523018867925  <span class="m">0</span>.555471698113  <span class="m">1</span>.97056603774
<span class="m">19</span>  <span class="m">65821</span>   <span class="m">66416</span>   <span class="m">0</span>.411764705882  <span class="m">3</span>.0     <span class="m">0</span>.636974789916  <span class="m">0</span>.168067226891  <span class="m">1</span>.67226890756
<span class="m">19</span>  <span class="m">65821</span>   <span class="m">70945</span>   <span class="m">0</span>.844600775761  <span class="m">4</span>.79176424668   <span class="m">31</span>.1346604215   <span class="m">0</span>.693073728066  <span class="m">1</span>.47911787666
<span class="m">19</span>  <span class="m">66319</span>   <span class="m">66492</span>   <span class="m">0</span>.774566473988  <span class="m">1</span>.59537572254   <span class="m">0</span>.0     <span class="m">0</span>.0     <span class="m">0</span>.578034682081
<span class="m">19</span>  <span class="m">66345</span>   <span class="m">71535</span>   <span class="m">0</span>.877430197151  <span class="m">5</span>.49036608863   <span class="m">43</span>.978805395    <span class="m">0</span>.746026011561  <span class="m">1</span>.43545279383
</pre></div>
</div>
<p>The default output of <code class="docutils literal notranslate"><span class="pre">multiBamSummary</span></code> (a compressed <code class="docutils literal notranslate"><span class="pre">numpy</span></code> array: <cite>*.npz</cite>) can be visualized using <a class="reference internal" href="plotCorrelation.html"><span class="doc">plotCorrelation</span></a> or <a class="reference internal" href="plotPCA.html"><span class="doc">plotPCA</span></a>.</p>
<p>The optional output (<code class="docutils literal notranslate"><span class="pre">--outRawCounts</span></code>) is a simple tab-delimited file that can be used with any other program. The first three columns define the region of the genome for which the reads were summarized.</p>
</section>
<section id="multibigwigsummary-in-galaxy">
<h2><a class="toc-backref" href="#id2">multiBigwigSummary in Galaxy</a><a class="headerlink" href="#multibigwigsummary-in-galaxy" title="Permalink to this headline"></a></h2>
<p>Below is the screenshot showing how to use multiBigwigSummary on the deeptools galaxy.</p>
<img alt="images/bigwiCorr_galaxy.png" src="images/bigwiCorr_galaxy.png" />
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Zilch.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>