<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>plotCorrelation &mdash; InfoScan 0.1 documentation</title>
      <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../../_static/css/custom-style.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
        <script src="../../_static/jquery.js"></script>
        <script src="../../_static/underscore.js"></script>
        <script src="../../_static/doctools.js"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../../index.html" class="icon icon-home"> InfoScan
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../InfoUpload.html">InfoUpload</a></li>
<li class="toctree-l1"><a class="reference internal" href="../NovelScan.html">NovelScan</a></li>
<li class="toctree-l1"><a class="reference internal" href="../CellInfo.html">CellInfo</a></li>
<li class="toctree-l1"><a class="reference internal" href="../about.html">About</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">InfoScan</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>plotCorrelation</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/content/tools/plotCorrelation.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="plotcorrelation">
<h1>plotCorrelation<a class="headerlink" href="#plotcorrelation" title="Permalink to this headline"></a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#background" id="id1">Background</a></p>
<ul>
<li><p><a class="reference internal" href="#correlation-calculation" id="id2">Correlation calculation</a></p></li>
<li><p><a class="reference internal" href="#hierarchical-clustering" id="id3">Hierarchical clustering</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#examples" id="id4">Examples</a></p></li>
</ul>
</div>
<section id="background">
<h2><a class="toc-backref" href="#id1">Background</a><a class="headerlink" href="#background" title="Permalink to this headline"></a></h2>
<p><code class="docutils literal notranslate"><span class="pre">plotCorrelation</span></code> computes the overall similarity between two or more files based on read coverage (or other scores) within genomic regions, which must be calculated using either <a class="reference internal" href="multiBamSummary.html"><span class="doc">multiBamSummary</span></a> or <a class="reference internal" href="multiBigwigSummary.html"><span class="doc">multiBigwigSummary</span></a>.</p>
<section id="correlation-calculation">
<h3><a class="toc-backref" href="#id2">Correlation calculation</a><a class="headerlink" href="#correlation-calculation" title="Permalink to this headline"></a></h3>
<p>The result of the correlation computation is a <strong>table of correlation coefficients</strong> that indicates how “strong” the relationship between two samples is and it will consist of numbers between -1 and 1. (-1 indicates perfect anti-correlation, 1 perfect correlation.)</p>
<img alt="images/QC_bamCorrelate_intro.png" src="images/QC_bamCorrelate_intro.png" />
<p>We offer two different functions for the correlation computation: <em>Pearson</em> or <em>Spearman</em>.</p>
<p>The <em>Pearson method</em> measures the <strong>metric differences</strong> between samples and is therefore influenced by outliers. More precisely, it is defined as the covariance of two variables divided by the product of their standard deviation.</p>
<p>The <em>Spearman method</em> is based on <strong>rankings</strong>.
If you imagine a race with 3 participants where the winner and runner-up are very close together while the third person broke her leg and comes in way, way after the first two, then Pearson would be strongly influenced by the fact that the third person had a great distance to the first ones while Spearman would only care about the fact that person 1 came in first, person 2 came in second and person 3 got the third rank, the distances between them are ignored.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Pearson is an appropriate measure for data that follows a normal distribution, while Spearman does not make this assumption and is generally less driven by outliers, but with the caveat of also being less sensitive.</p>
</div>
</section>
<section id="hierarchical-clustering">
<h3><a class="toc-backref" href="#id3">Hierarchical clustering</a><a class="headerlink" href="#hierarchical-clustering" title="Permalink to this headline"></a></h3>
<p>If you use the heatmap output of <code class="docutils literal notranslate"><span class="pre">plotCorrelation</span></code>, this will automatically lead to a clustering of the samples based on the correlation coefficients. This helps to determine whether the different sample types can be separated, i.e., samples of different conditions are expected to be more dissimilar to each other than replicates within the same condition.</p>
<p>The <em>distances</em> of the sample pairs are based on the correlation coefficients, <em>r</em>, where distance = 1 - <em>r</em>. The similarity of the samples is assessed using the nearest point algorithm, i.e., the shortest distance between any 2 members of the tree is considered to decide whether to join a cluster or not. For more details of the algorithm, go <a class="reference external" href="http://docs.scipy.org/doc/scipy-0.16.0/reference/generated/scipy.cluster.hierarchy.linkage.html">here</a>.</p>
</section>
</section>
<section id="examples">
<h2><a class="toc-backref" href="#id4">Examples</a><a class="headerlink" href="#examples" title="Permalink to this headline"></a></h2>
<p>Here’s an example of RNA-seq data from different human cell lines that we had downloaded from <a class="reference external" href="https://genome.ucsc.edu/ENCODE/dataMatrix/encodeDataMatrixHuman.html">https://genome.ucsc.edu/ENCODE/dataMatrix/encodeDataMatrixHuman.html</a>.</p>
<img alt="images/QC_bamCorrelate_RNAseq.png" src="images/QC_bamCorrelate_RNAseq.png" />
<p>As you can see, both correlation calculations more or less agree on which samples are nearly identical (the replicates, indicated by 1 or 2 at the end of the label). The Spearman correlation, however, seems to be more robust and meets our expectations more closely as the two different cell types (HUVEC and IMR90) are clearly separated.</p>
<p>In the following example, a correlation analysis is performed based on the coverage file computed by <a class="reference internal" href="multiBamSummary.html"><span class="doc">multiBamSummary</span></a> or <a class="reference internal" href="multiBigwigSummary.html"><span class="doc">multiBigwigSummary</span></a> for our test ENCODE ChIP-Seq datasets.</p>
<p><strong>Scatterplot</strong></p>
<p>Here we make pairwose scatterplots of the average scores per transcript that we calculated using <a class="reference internal" href="multiBigwigSummary.html"><span class="doc">multiBigwigSummary</span></a> and include the Pearson correlation coefficients for each comparison.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ deepTools2.0/bin/plotCorrelation <span class="se">\</span>
-in scores_per_transcript.npz <span class="se">\</span>
--corMethod pearson --skipZeros <span class="se">\</span>
--plotTitle <span class="s2">&quot;Pearson Correlation of Average Scores Per Transcript&quot;</span> <span class="se">\</span>
--whatToPlot scatterplot <span class="se">\</span>
-o scatterplot_PearsonCorr_bigwigScores.png   <span class="se">\</span>
--outFileCorMatrix PearsonCorr_bigwigScores.tab
</pre></div>
</div>
<img alt="images/test_plots/scatterplot_PearsonCorr_bigwigScores.png" src="images/test_plots/scatterplot_PearsonCorr_bigwigScores.png" />
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ cat PearsonCorr_bigwigScores.tab
    <span class="s1">&#39;H3K27me3&#39;</span>      <span class="s1">&#39;H3K4me1&#39;</span>       <span class="s1">&#39;H3K4me3&#39;</span>       <span class="s1">&#39;HeK9me3&#39;</span>       <span class="s1">&#39;input&#39;</span>
    <span class="s1">&#39;H3K27me3&#39;</span>      <span class="m">1</span>.0000  -0.1032 -0.1269 -0.0339 -0.0395
    <span class="s1">&#39;H3K4me1&#39;</span>       -0.1032 <span class="m">1</span>.0000  <span class="m">0</span>.3985  -0.1863 <span class="m">0</span>.3328
    <span class="s1">&#39;H3K4me3&#39;</span>       -0.1269 <span class="m">0</span>.3985  <span class="m">1</span>.0000  -0.0480 <span class="m">0</span>.2822
    <span class="s1">&#39;HeK9me3&#39;</span>       -0.0339 -0.1863 -0.0480 <span class="m">1</span>.0000  -0.0353
    <span class="s1">&#39;input&#39;</span> -0.0395 <span class="m">0</span>.3328  <span class="m">0</span>.2822  -0.0353 <span class="m">1</span>.0000
</pre></div>
</div>
<p><strong>Heatmap</strong></p>
<p>In addition to scatterplots, heatmaps can be generated where the pairwise correlation coefficients are depicted by varying color intensities and are clustered using hierarchical clustering.</p>
<p>The example here calculates the Spearman correlation coefficients of read counts.
The dendrogram indicates which samples’ read counts are most similar to each other.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ deepTools2.0/bin/plotCorrelation <span class="se">\</span>
    -in readCounts.npz <span class="se">\</span>
    --corMethod spearman --skipZeros <span class="se">\</span>
    --plotTitle <span class="s2">&quot;Spearman Correlation of Read Counts&quot;</span> <span class="se">\</span>
    --whatToPlot heatmap --colorMap RdYlBu --plotNumbers <span class="se">\</span>
    -o heatmap_SpearmanCorr_readCounts.png   <span class="se">\</span>
    --outFileCorMatrix SpearmanCorr_readCounts.tab
</pre></div>
</div>
<img alt="images/test_plots/heatmap_SpearmanCorr_readCounts.png" src="images/test_plots/heatmap_SpearmanCorr_readCounts.png" />
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Zilch.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>