<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Changes in deepTools2.0 &mdash; InfoScan 0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/custom-style.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> InfoScan
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="InfoUpload.html">InfoUpload</a></li>
<li class="toctree-l1"><a class="reference internal" href="NovelScan.html">NovelScan</a></li>
<li class="toctree-l1"><a class="reference internal" href="CellInfo.html">CellInfo</a></li>
<li class="toctree-l1"><a class="reference internal" href="about.html">About</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">InfoScan</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Changes in deepTools2.0</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/changelog.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="changes-in-deeptools2-0">
<h1>Changes in deepTools2.0<a class="headerlink" href="#changes-in-deeptools2-0" title="Permalink to this headline"></a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#major-changes" id="id1">Major changes</a></p>
<ul>
<li><p><a class="reference internal" href="#accommodating-additional-data-types" id="id2">Accommodating additional data types</a></p></li>
<li><p><a class="reference internal" href="#structural-updates" id="id3">Structural updates</a></p></li>
<li><p><a class="reference internal" href="#renamed-tools" id="id4">Renamed tools</a></p></li>
<li><p><a class="reference internal" href="#increased-efficiency" id="id5">Increased efficiency</a></p></li>
<li><p><a class="reference internal" href="#new-features-and-tools" id="id6">New features and tools</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#minor-changes" id="id7">Minor changes</a></p>
<ul>
<li><p><a class="reference internal" href="#changed-parameters-names-and-settings" id="id8">Changed parameters names and settings</a></p></li>
<li><p><a class="reference internal" href="#bug-fixes" id="id9">Bug fixes</a></p></li>
</ul>
</li>
</ul>
</div>
<section id="major-changes">
<h2><a class="toc-backref" href="#id1">Major changes</a><a class="headerlink" href="#major-changes" title="Permalink to this headline"></a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The major changes encompass features for <strong>increased efficiency</strong>, <strong>new sequencing data types</strong>, and <strong>additional plots</strong>, particularly for QC.</p>
</div>
<p>Moreover, deepTools modules can now be used by other python programs.
The <a class="reference internal" href="example_api_tutorial.html#api"><span class="std std-ref">deepTools API example</span></a> is part of the new documentation.</p>
<section id="accommodating-additional-data-types">
<h3><a class="toc-backref" href="#id2">Accommodating additional data types</a><a class="headerlink" href="#accommodating-additional-data-types" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p>correlation and comparisons can now be calculated for <strong>bigWig files</strong> (in addition to BAM files) using <code class="docutils literal notranslate"><span class="pre">multiBigwigSummary</span></code> and <code class="docutils literal notranslate"><span class="pre">bigwigCompare</span></code></p></li>
<li><p><strong>RNA-seq:</strong> split-reads are now natively supported</p></li>
<li><p><strong>MNase-seq:</strong> using the new option <code class="docutils literal notranslate"><span class="pre">--MNase</span></code> in <code class="docutils literal notranslate"><span class="pre">bamCoverage</span></code>, one can now compute read coverage only taking the 2 central base pairs of each mapped fragment into account.</p></li>
</ul>
</section>
<section id="structural-updates">
<h3><a class="toc-backref" href="#id3">Structural updates</a><a class="headerlink" href="#structural-updates" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p>All modules have comprehensive and automatic tests that evaluate proper functioning after any modification of the code.</p></li>
<li><p>Virtualization for stability: we now provide a <code class="docutils literal notranslate"><span class="pre">docker</span></code> image and enable the easy deployment of deepTools via the Galaxy <code class="docutils literal notranslate"><span class="pre">toolshed</span></code>.</p></li>
<li><p>Our documentation is now version-aware thanks to readthedocs and <code class="docutils literal notranslate"><span class="pre">sphinx</span></code>.</p></li>
<li><p>The API is public and documented.</p></li>
</ul>
</section>
<section id="renamed-tools">
<h3><a class="toc-backref" href="#id4">Renamed tools</a><a class="headerlink" href="#renamed-tools" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p><strong>heatmapper</strong> to <a class="reference internal" href="tools/plotHeatmap.html"><span class="doc">plotHeatmap</span></a></p></li>
<li><p><strong>profiler</strong> to <a class="reference internal" href="tools/plotProfile.html"><span class="doc">plotProfile</span></a></p></li>
<li><p><strong>bamCorrelate</strong> to <a class="reference internal" href="tools/multiBamSummary.html"><span class="doc">multiBamSummary</span></a></p></li>
<li><p><strong>bigwigCorrelate</strong> to <a class="reference internal" href="tools/multiBigwigSummary.html"><span class="doc">multiBigwigSummary</span></a></p></li>
<li><p><strong>bamFingerprint</strong> to <a class="reference internal" href="tools/plotFingerprint.html"><span class="doc">plotFingerprint</span></a></p></li>
</ul>
</section>
<section id="increased-efficiency">
<h3><a class="toc-backref" href="#id5">Increased efficiency</a><a class="headerlink" href="#increased-efficiency" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p>We dramatically improved the <strong>speed</strong> of bigwig related tools (<a class="reference internal" href="tools/multiBigwigSummary.html"><span class="doc">multiBigwigSummary</span></a> and <code class="docutils literal notranslate"><span class="pre">computeMatrix</span></code>) by using the new <a class="reference external" href="https://github.com/dpryan79/pyBigWig">pyBigWig module</a>.</p></li>
<li><p>It is now possible to generate one composite heatmap and/or meta-gene image based on <strong>multiple bigwig files</strong> in one go (see <a class="reference internal" href="tools/computeMatrix.html"><span class="doc">computeMatrix</span></a>, <a class="reference internal" href="tools/plotHeatmap.html"><span class="doc">plotHeatmap</span></a>, and <a class="reference internal" href="tools/plotProfile.html"><span class="doc">plotProfile</span></a> for examples)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">computeMatrix</span></code> now also accepts multiple input BED files. Each is treated as a group within a sample and is plotted independently.</p></li>
<li><p>We added <strong>additional filtering options for handling BAM files</strong>, decreasing the need for prior filtering using tools other than deepTools: The <code class="docutils literal notranslate"><span class="pre">--samFlagInclude</span></code> and <code class="docutils literal notranslate"><span class="pre">--samFlagExclude</span></code> parameters can, for example, be used to only include (or exclude) forward reads in an analysis.</p></li>
<li><p>We separated the generation of read count tables from the calculation of pairwise correlations that was previously handled by <code class="docutils literal notranslate"><span class="pre">bamCorrelate</span></code>. Now, read counts are calculated first using <code class="docutils literal notranslate"><span class="pre">multiBamSummary</span></code> or <code class="docutils literal notranslate"><span class="pre">multiBigWigCoverage</span></code> and the resulting output file can be used for calculating and plotting pairwise correlations using <code class="docutils literal notranslate"><span class="pre">plotCorrelation</span></code> or for doing a principal component analysis using <code class="docutils literal notranslate"><span class="pre">plotPCA</span></code>.</p></li>
</ul>
</section>
<section id="new-features-and-tools">
<h3><a class="toc-backref" href="#id6">New features and tools</a><a class="headerlink" href="#new-features-and-tools" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p>Correlation analyses are no longer limited to BAM files – bigwig files are possible, too! (see <a class="reference internal" href="tools/multiBigwigSummary.html"><span class="doc">multiBigwigSummary</span></a>)</p></li>
<li><p>Correlation coefficients can now be computed even if the data contains NaNs.</p></li>
<li><dl class="simple">
<dt>Added <strong>new quality control</strong> tools:</dt><dd><ul>
<li><p>use <a class="reference internal" href="tools/plotCoverage.html"><span class="doc">plotCoverage</span></a> to plot the coverage over base pairs</p></li>
<li><p>use <a class="reference internal" href="tools/plotPCA.html"><span class="doc">plotPCA</span></a> for principal component analysis</p></li>
<li><p><a class="reference internal" href="tools/bamPEFragmentSize.html"><span class="doc">bamPEFragmentSize</span></a> can be used to calculate the average fragment size for paired-end read data</p></li>
</ul>
</dd>
</dl>
</li>
<li><p>Added the possibility for <strong>hierarchical clustering</strong>, besides <em>k</em>-means to <code class="docutils literal notranslate"><span class="pre">plotProfile</span></code> and <code class="docutils literal notranslate"><span class="pre">plotHeatmap</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">plotProfile</span></code> has many more options to make compelling summary plots</p></li>
</ul>
</section>
</section>
<section id="minor-changes">
<h2><a class="toc-backref" href="#id7">Minor changes</a><a class="headerlink" href="#minor-changes" title="Permalink to this headline"></a></h2>
<section id="changed-parameters-names-and-settings">
<h3><a class="toc-backref" href="#id8">Changed parameters names and settings</a><a class="headerlink" href="#changed-parameters-names-and-settings" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">computeMatrix</span></code> can now read files with DOS newline characters.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--missingDataAsZero</span></code> was renamed to <code class="docutils literal notranslate"><span class="pre">--skipNonCoveredRegions</span></code> for clarity in <code class="docutils literal notranslate"><span class="pre">bamCoverage</span></code> and <code class="docutils literal notranslate"><span class="pre">bamCompare</span></code>.</p></li>
<li><p>Read extension was made optional and we removed the need to specify a default fragment length for most of the tools: <code class="docutils literal notranslate"><span class="pre">--fragmentLength</span></code> was thus replaced by the new optional parameter <code class="docutils literal notranslate"><span class="pre">--extendReads</span></code>.</p></li>
<li><p>Added option <code class="docutils literal notranslate"><span class="pre">--skipChromosomes</span></code> to <code class="docutils literal notranslate"><span class="pre">multiBigwigSummary</span></code>, which can be used to, for example, skip all ‘random’ chromosomes.</p></li>
<li><p>Added the option for adding titles to QC plots.</p></li>
</ul>
</section>
<section id="bug-fixes">
<h3><a class="toc-backref" href="#id9">Bug fixes</a><a class="headerlink" href="#bug-fixes" title="Permalink to this headline"></a></h3>
<ul class="simple">
<li><p>Resolved an error introduced by <code class="docutils literal notranslate"><span class="pre">numpy</span> <span class="pre">version</span> <span class="pre">1.10</span></code> in <code class="docutils literal notranslate"><span class="pre">computeMatrix</span></code>.</p></li>
<li><p>Improved plotting features for <code class="docutils literal notranslate"><span class="pre">plotProfile</span></code> when using as plot type: ‘overlapped_lines’ and ‘heatmap’</p></li>
<li><p>Fixed problem with BED intervals in <code class="docutils literal notranslate"><span class="pre">multiBigwigSummary</span></code> and <code class="docutils literal notranslate"><span class="pre">multiBamSummary</span></code> that returned wrongly labeled raw counts.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">multiBigwigSummary</span></code> now also considers chromosomes as identical when the names between samples differ by ‘chr’ prefix, e.g. chr1 vs. 1.</p></li>
<li><p>Fixed problem with wrongly labeled proper read pairs in a BAM file. We now have additional checks to determine if a read pair is a proper pair: the reads must face each other and are not allowed to be farther apart than 4x the mean fragment length.</p></li>
<li><p>For <code class="docutils literal notranslate"><span class="pre">bamCoverage</span></code> and <code class="docutils literal notranslate"><span class="pre">bamCompare</span></code>, the behavior of <code class="docutils literal notranslate"><span class="pre">scaleFactor</span></code> was updated such that now, if given in combination with the normalization options (<code class="docutils literal notranslate"><span class="pre">--normalizeTo1x</span></code> or <code class="docutils literal notranslate"><span class="pre">--normalizeUsingRPKM</span></code>), the given scaling factor will be multiplied with the factor computed by the respective normalization method.</p></li>
</ul>
</section>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Zilch.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>