<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Using deepTools within Galaxy &mdash; InfoScan 0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/custom-style.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> InfoScan
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="InfoUpload.html">InfoUpload</a></li>
<li class="toctree-l1"><a class="reference internal" href="NovelScan.html">NovelScan</a></li>
<li class="toctree-l1"><a class="reference internal" href="CellInfo.html">CellInfo</a></li>
<li class="toctree-l1"><a class="reference internal" href="about.html">About</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">InfoScan</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Using deepTools within Galaxy</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/help_galaxy_intro.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="using-deeptools-within-galaxy">
<h1>Using deepTools within Galaxy<a class="headerlink" href="#using-deeptools-within-galaxy" title="Permalink to this headline"></a></h1>
<p><a class="reference external" href="http://galaxyproject.org/">Galaxy</a> is a tremendously useful platform developed by the Galaxy Team at Penn State and the Emory University. This platform is meant to offer access to a large variety of bioinformatics tools that can be used without computer programming experiences. That means, that the basic features of Galaxy will apply to every tool, i.e. every tool provided within a Galaxy framework will look very similar and will follow the concepts of Galaxy.</p>
<p>Our publicly available deepTools Galaxy instance can be found here:
<a class="reference external" href="http://deeptools.ie-freiburg.mpg.de/">deeptools.ie-freiburg.mpg.de</a>.
This server also contains some additional tools that will enable users to analyse and visualize data from high-throughput sequencing experiments, starting from aligned reads.</p>
<p><strong>Table of content</strong></p>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#basic-features-of-galaxy" id="id1">Basic features of Galaxy</a></p>
<ul>
<li><p><a class="reference internal" href="#the-start-site" id="id2">The start site</a></p></li>
<li><p><a class="reference internal" href="#details" id="id3">Details</a></p></li>
<li><p><a class="reference internal" href="#handling-failed-files" id="id4">Handling failed files</a></p></li>
<li><p><a class="reference internal" href="#workflows" id="id5">Workflows</a></p></li>
</ul>
</li>
</ul>
</div>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="help_galaxy_dataup.html">Data import into Galaxy</a><ul>
<li class="toctree-l2"><a class="reference internal" href="help_galaxy_dataup.html#upload-files-from-your-computer">Upload files from your computer</a></li>
<li class="toctree-l2"><a class="reference internal" href="help_galaxy_dataup.html#import-data-sets-from-the-galaxy-data-library">Import data sets from the Galaxy data library</a></li>
<li class="toctree-l2"><a class="reference internal" href="help_galaxy_dataup.html#download-annotation-files-from-public-data-bases">Download annotation files from public data bases</a></li>
<li class="toctree-l2"><a class="reference internal" href="help_galaxy_dataup.html#copy-data-sets-between-histories">Copy data sets between histories</a></li>
</ul>
</li>
</ul>
</div>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="help_galaxy_deeptools.html">Which tools can I find in the deepTools Galaxy?</a><ul>
<li class="toctree-l2"><a class="reference internal" href="help_galaxy_deeptools.html#deeptools">deepTools</a></li>
<li class="toctree-l2"><a class="reference internal" href="help_galaxy_deeptools.html#working-with-text-files-and-tables">Working with text files and tables</a></li>
<li class="toctree-l2"><a class="reference internal" href="help_galaxy_deeptools.html#basic-arithmetics-for-tables">Basic arithmetics for tables</a></li>
</ul>
</li>
</ul>
</div>
<section id="basic-features-of-galaxy">
<h2><a class="toc-backref" href="#id1">Basic features of Galaxy</a><a class="headerlink" href="#basic-features-of-galaxy" title="Permalink to this headline"></a></h2>
<p>Galaxy is a web-based platform for data intensive, bioinformatics-dependent research and it is being developed by Penn State and John Hopkins University. The original Galaxy can be found <a class="reference external" href="https://galaxyproject.org/">here</a>.</p>
<p>Since it is impossible to meet all bioinformatics needs – that can range from evolutionary analysis to data from mass spectrometry to high-throughput DNA sequencing (and way beyond) – with one single web server, many institutes have installed their own versions of the Galaxy platform tuned to their specific needs.</p>
<p>Our <a class="reference external" href="http://deeptools.ie-freiburg.mpg.de/">deepTools Galaxy</a> is such a specialized server dedicated to the analysis of high-throughput DNA sequencing data. The overall makeup of this web server, however, is the same as for any other Galaxy installation, so if you’ve used Galaxy before, you will learn to use deepTools in no time!</p>
<section id="the-start-site">
<h3><a class="toc-backref" href="#id2">The start site</a><a class="headerlink" href="#the-start-site" title="Permalink to this headline"></a></h3>
<p>Here is a screenshot of what the start site will approximately look like:</p>
<img alt="images/Gal_startsite.png" src="images/Gal_startsite.png" />
<p>The start site contains 4 main features:</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 13%" />
<col style="width: 87%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><strong>Top menu</strong></p></td>
<td><p>Your gateway away from the actual analysis part into other sections of Galaxy, e.g. <strong>workflows</strong> and <strong>shared data</strong>.</p></td>
</tr>
<tr class="row-even"><td><p><strong>Tool panel</strong></p></td>
<td><p><em>What can be done?</em> Find all tools installed in this Galaxy instance</p></td>
</tr>
<tr class="row-odd"><td><p><strong>Main frame</strong></p></td>
<td><p><em>What am I doing?</em> This is your main <strong>working space</strong> where input will be required from you once you’ve selected a tool.</p></td>
</tr>
<tr class="row-even"><td><p><strong>History panel</strong></p></td>
<td><p><em>What did I do?</em> The history is like a <strong>log book</strong>: everything you ever did is recorded here.</p></td>
</tr>
</tbody>
</table>
<p>For those visual learners, here’s an annotated screenshot:</p>
<img alt="images/Gal_startsite_with_comments.png" src="images/Gal_startsite_with_comments.png" />
</section>
<section id="details">
<h3><a class="toc-backref" href="#id3">Details</a><a class="headerlink" href="#details" title="Permalink to this headline"></a></h3>
<p>In the default state of the tool panel you see the <strong>tool categories</strong>, e.g. “Get Data”. If you click on them, you will see the <strong>individual tools</strong> belonging to each category, e.g. “Upload File from your computer”, “UCSC Main table browser” and “Biomart central server” in case you clicked on “Get Data”. To use a tool such as “Upload File from your computer”, just click on it.</p>
<p>The <strong>tool *search* panel</strong> is extremely useful as it allows you to enter a key word (e.g. “bam”) that will lead to all the tools mentioning the key word in the tool name.</p>
<p>Once you’ve uploaded any kind of data, you will find the history on the
right hand side filling up with green tiles.
Each tile corresponds to one data set that you either uploaded or created.
The data sets can be images, raw sequencing files, text files, tables - virtually anything.
The content of a data set <em>cannot</em> be modified - every time you want to change something <em>within</em> a data file (e.g. you would like to sort the values or add a line or cut a column), you will have to use a Galaxy tool that will lead to a <em>new</em> data set being produced.
This behaviour is often confusing for Galaxy novices (as histories tend to accumulate data sets very quickly), but it is necessary to enforce the strict policy of documenting <em>every modification</em> to a given data set.
Eventhough your history might be full of data sets with strange names, you will always be able to track back the source and evolution of each file.
Also, every data set can be downloaded to your computer individually.
Alternatively, you can <em>download</em> an entire history or <em>share</em> the history with another user.</p>
<p>Have a look at the following screenshot to get a feeling for how many information Galaxy keeps for you (which makes it very feasible to reproduce any analysis):</p>
<img alt="images/Gal_screenshot_dataSet.png" src="images/Gal_screenshot_dataSet.png" />
<p>Each data set can have 4 different states that are intuitively color-coded:</p>
<img alt="images/Gal_screenshot_dataSetStates.png" src="images/Gal_screenshot_dataSetStates.png" />
</section>
<section id="handling-failed-files">
<h3><a class="toc-backref" href="#id4">Handling failed files</a><a class="headerlink" href="#handling-failed-files" title="Permalink to this headline"></a></h3>
<blockquote>
<div><p>If you encounter a failed file after you’ve run a tool, please do the following steps (<strong>in this order</strong>):</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>click on the center button on the lower left corner of the failed data set <code class="docutils literal notranslate"><span class="pre">(i)</span></code>: did you chose the <strong>correct data files</strong>?</p></li>
<li><p>if you’re sure that you chose the correct files, hit the <code class="docutils literal notranslate"><span class="pre">re-run</span> <span class="pre">button</span></code> (blue arrow in the lower left corner) - check again whether your files had the <strong>correct file format</strong>. If you suspect that the format might be incorrectly assigned (e.g. a file that should be a BED file is labelled as a tabular file), click the <code class="docutils literal notranslate"><span class="pre">edit</span> <span class="pre">button</span></code> (the pencil) of the input data file - there you can change the corresponding attributes</p></li>
<li><p>if you’ve checked your input data and the error is persisting, click on the <code class="docutils literal notranslate"><span class="pre">green</span> <span class="pre">bug</span></code> (lower left corner of the failed data set) and send the <strong>bug report</strong> to us. You do not need to indicate a valid email-address unless you would like us to get in touch with you once the issue is solved.</p></li>
</ol>
</div></blockquote>
</div></blockquote>
</section>
<section id="workflows">
<h3><a class="toc-backref" href="#id5">Workflows</a><a class="headerlink" href="#workflows" title="Permalink to this headline"></a></h3>
<p>Workflows are Galaxy’s equivalent of protocols.
This is a very useful feature as it allows users to <em>share their protocols and bioinformatic analyses</em> in a very easy and transparent way.
This is the graphical representation of a Galaxy workflow that can easily be modified via drag’n’drop within the workflows manual (you must be registered with deepTools Galaxy to be able to generate your own workflows or edit published ones).</p>
<img alt="images/Gal_workflow.png" src="images/Gal_workflow.png" />
<p><strong>More help</strong></p>
<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>If you encounter a failing data set (marked in red), please <strong>send a bug report</strong> via the Galaxy bug report button and we will get in touch if you indicate your email address.</p>
</div>
<table class="docutils align-default">
<colgroup>
<col style="width: 55%" />
<col style="width: 45%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference external" href="http://wiki.galaxyproject.org/Learn">http://wiki.galaxyproject.org/Learn</a></p></td>
<td><p>Help for Galaxy usage in general</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="Galaxy-related-FAQs">deepTools Galaxy FAQs</a></p></td>
<td><p>Frequently encountered issues with our specific Galaxy instance</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="http://biostars.org">Biostars</a></p></td>
<td><p>For issues not addressed in the FAQs</p></td>
</tr>
</tbody>
</table>
</section>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Zilch.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>