<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Data import into Galaxy &mdash; InfoScan 0.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/custom-style.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> InfoScan
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="InfoUpload.html">InfoUpload</a></li>
<li class="toctree-l1"><a class="reference internal" href="NovelScan.html">NovelScan</a></li>
<li class="toctree-l1"><a class="reference internal" href="CellInfo.html">CellInfo</a></li>
<li class="toctree-l1"><a class="reference internal" href="about.html">About</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">InfoScan</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Data import into Galaxy</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/help_galaxy_dataup.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="data-import-into-galaxy">
<h1>Data import into Galaxy<a class="headerlink" href="#data-import-into-galaxy" title="Permalink to this headline"></a></h1>
<p>There are three main ways to populate your Galaxy history with data
files plus an additional one for sharing data within Galaxy.</p>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#upload-files-from-your-computer" id="id1">Upload files from your computer</a></p></li>
<li><p><a class="reference internal" href="#import-data-sets-from-the-galaxy-data-library" id="id2">Import data sets from the Galaxy data library</a></p></li>
<li><p><a class="reference internal" href="#download-annotation-files-from-public-data-bases" id="id3">Download annotation files from public data bases</a></p></li>
<li><p><a class="reference internal" href="#copy-data-sets-between-histories" id="id4">Copy data sets between histories</a></p></li>
</ul>
</div>
<section id="upload-files-from-your-computer">
<h2><a class="toc-backref" href="#id1">Upload files from your computer</a><a class="headerlink" href="#upload-files-from-your-computer" title="Permalink to this headline"></a></h2>
<p>The data upload of files <strong>smaller than 2 GB</strong> that lie on your computer is fairly straight-forward: click on the category “Get data” and choose the tool “Upload file”.
Then select the file via the “Browse” button.</p>
<img alt="images/Gal_DataUpload.png" src="images/Gal_DataUpload.png" />
<p>For files <strong>greater than 2GB</strong>, there’s the option to upload via an FTP server. If your data is available via an URL that links to an FTP server, you can simply
paste the URL in the empty text box.</p>
<p>If you do not have access to an FTP server, you can directly upload to
our Galaxy’s FTP.</p>
<ol class="arabic simple">
<li><p>Register with deeptools.ie-freiburg.mpg.de (via “User” ⟶ “register”; registration requires an email address and is free of charge)</p></li>
<li><p>You will also need an FTP client, e.g. <a class="reference external" href="https://filezilla-project.org/">filezilla</a>.</p></li>
<li><p>Then login to the <strong>FTP client</strong> using your <strong>deepTools Galaxy user name and password</strong> (host: deeptools.ie-freiburg.mpg.de). Down below you see a screenshot of what that looks like with filezilla.</p></li>
<li><p>Copy the file you wish to upload to the remote site (in filezilla, you can simply drag the file to the window on the right hand side)</p></li>
<li><p>Go back to <a class="reference external" href="http://deeptools.ie-freiburg.mpg.de/">deepTools Galaxy</a>.</p></li>
<li><p>Click on the tool “Upload file” (⟶ “Files uploaded via FTP”) - here, the files you just copied over via filezilla should appear. Select the files you want and hit “execute”. They will be moved from the FTP server to your history (i.e. they will be deleted from the FTP once the upload was successful).</p></li>
</ol>
<img alt="images/Gal_filezilla.png" src="images/Gal_filezilla.png" />
</section>
<section id="import-data-sets-from-the-galaxy-data-library">
<h2><a class="toc-backref" href="#id2">Import data sets from the Galaxy data library</a><a class="headerlink" href="#import-data-sets-from-the-galaxy-data-library" title="Permalink to this headline"></a></h2>
<p>If you would like to play around with sample data, you can import files
that we have saved within the general data storage of the deepTools
Galaxy server. Everyone can import them into his or her own history,
they will not contribute to the user’s disk quota.</p>
<p>You can reach the data library via “Shared Data” in the top menu, then
select “Data Libraries”.</p>
<p>Within the Data Library you will find a folder called “Sample Data” that
contains data that we downloaded from the <a class="reference external" href="http://www.roadmapepigenomics.org/data">Roadmap
project</a> and
<a class="reference external" href="http://genome.ucsc.edu/">UCSC</a>
More precisely, we donwloaded the [FASTQ][] files of various ChIP-seq samples and the corresponding input and mapped the reads to the human reference genome (version hg19) to obtain the [BAM][] files you see.
In addition, you will find bigWig files created using <code class="docutils literal notranslate"><span class="pre">bamCoverage</span></code> and some annotation files in BED format as well as RNA-seq data.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>To keep the file size smallish, all files contain data for chromosome 19 and chromosome X only!</p>
</div>
<img alt="images/Gal_DataLib.png" src="images/Gal_DataLib.png" />
</section>
<section id="download-annotation-files-from-public-data-bases">
<h2><a class="toc-backref" href="#id3">Download annotation files from public data bases</a><a class="headerlink" href="#download-annotation-files-from-public-data-bases" title="Permalink to this headline"></a></h2>
<p>In many cases you will want to query your sequencing data results for
known genome annotation, such as genes, exons, transcription start sites
etc. These information can be obtained via the two main sources of
genome annotation, <a class="reference external" href="http://genome.ucsc.edu/">UCSC</a> and <a class="reference external" href="http://www.biomart.org/">BioMart</a>.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>UCSC and BioMart cater to different ways of genome annotation, i.e. genes defined in UCSC might not correspond to the same regions in a gene file downloaded from BioMart. (For a brief overview over the issues of genome annotation, you can check out <a class="reference external" href="http://en.wikipedia.org/wiki/Genome_project">Wikipedia</a>, if you always wanted to know much more about those issues, <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/22510764">this</a> might be a good start.)</p>
</div>
<p>You can access the data stored at UCSC or BioMart conveniently through our Galaxy instance which will import the resulting files into your history. Just go to <strong>“Get data”</strong> ⟶ “UCSC” or “BioMart”.</p>
<p>The majority of annotation files will probably be in [BED][] format, however, you can also find other data sets.
UCSC, for example, offers a wide range of data that you can browse via the “group” and “track” menus (for example, you could download the GC content of the genome as a signal file from UCSC via the “group” menu (“Mapping and Sequencing Tracks”).</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The download through this interface is limited to 100,000 lines per file which might not be sufficient for some mammalian data sets.</p>
</div>
<p>Here’s a screenshot from downloading a BED-file of all RefSeq genes defined for the human genome (version hg19):</p>
<img alt="images/Gal_UCSC.png" src="images/Gal_UCSC.png" />
<p>And here’s how you would do it for the BioMart approach:</p>
<img alt="images/Gal_biomart.png" src="images/Gal_biomart.png" />
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Per default, <strong>BioMart will not output a BED file</strong> like UCSC does. It is therefore important that you make sure you get all the information you need (most likely: chromosome, gene start, gene end, ID, strand) via the “Attributes” section. You can click on the “Results” button at any time to check the format of the table that will be sent to Galaxy (Note that the strand information will be decoded as 1 for “forward” or “plus” strand and -1 for “reverse” or “minus” strand).</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Be aware, that BED files from UCSC will have chromosomes labelled with “chr” while ENSEMBL usually returns just the number – this might lead to incompatibilities, i.e. when working with annotations from UCSC and ENSEMBL, you need to make sure to use the same naming!</p>
</div>
</section>
<section id="copy-data-sets-between-histories">
<h2><a class="toc-backref" href="#id4">Copy data sets between histories</a><a class="headerlink" href="#copy-data-sets-between-histories" title="Permalink to this headline"></a></h2>
<p>If you have registered with deepTools Galaxy you can have more than one history.</p>
<p>In order to minimize the disk space you’re occupying we strongly suggest to <strong>copy</strong> data sets between histories when you’re using the same data set in different histories.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Copying data sets is only possible for registered users.</p>
</div>
<img alt="images/Gal_copy.png" src="images/Gal_copy.png" />
<p>Copying can easily be done via the History panel’s <code class="docutils literal notranslate"><span class="pre">option</span></code> button ⟶ “Copy dataset”. In the main frame, you should now be able to select the history you would like to copy from on the left hand side and the target history on the right hand side.</p>
<p><strong>More help</strong></p>
<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>If you encounter a failing data set (marked in red), please <strong>send a bug report</strong> via the Galaxy bug report button and we will get in touch if you indicate your email address.</p>
</div>
<table class="docutils align-default">
<colgroup>
<col style="width: 55%" />
<col style="width: 45%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference external" href="http://wiki.galaxyproject.org/Learn">http://wiki.galaxyproject.org/Learn</a></p></td>
<td><p>Help for Galaxy usage in general</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference external" href="Galaxy-related-FAQs">deepTools Galaxy FAQs</a></p></td>
<td><p>Frequently encountered issues with our specific Galaxy instance</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference external" href="http://biostars.org">Biostars</a></p></td>
<td><p>For issues not addressed in the FAQs</p></td>
</tr>
</tbody>
</table>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Zilch.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>